### ISSUE TITLE
Assistance Requested for Fine-Tuning on Visual Question Answering Task

### ISSUE THREAD

**Issue by crux82** [THUDM/CogVLM#455]
First of all, thank you for this excellent work.

I would like to fine-tune the model from https://huggingface.co/THUDM/cogagent-vqa-hf on a visual question answering task. I would like to input a pair consisting of an image and a prompt (a question string), and I expect the output to be the answer (a string) and one or more bounding boxes with grounding to support the answer.

Unfortunately, I am unable to understand how the examples for training should be passed in the script

https://github.com/THUDM/CogVLM/blob/main/finetune_demo/finetune_cogagent_lora.sh

Could you kindly help me?

Thank you very much,

Danilo

**Comment by zRzRzRzRzRzRzR** [comment]
bbox 也需要按照 字符的输入和输出格式进行匹配，格式为 [x,y,w,h](如果您是为了微调），可以直接放在文字中输出，bbox并不需要单独的处理，直接放在常规文字的后方就行

这是一个例子：

图像中有两只猫，分别是[[x1,y,1,w1.h1],[x2,y2,w2,h2]]这种格式

**Comment by crux82** [comment]
@zRzRzRzRzRzRzR 

OK!

First of all, thank you very much for the prompt response.

One last question: from the script example with the dataset from https://www.kaggle.com/datasets/aadhavvignesh/captcha-images, only an image is used as input, without any additional text (in my case, the question), and the output is expected to be a string.

While the output is clear to me, I am now unclear on how to pass the image/text input pair.

Can you help me?

Thank you again!

**Comment by basteran** [comment]
Dear @zRzRzRzRzRzRzR and other authors, I am interested too in fine-tuning your model on a Visual Question Answering task. 

Looking through the repository and the scripts, I can't find any mention of how to format my bounding boxes in the labels. As an example, if I want the model to generate in output something like   
_"The girl [[000,111,222,333]] is standing tall and the boy [[000,111,222,333]] is sitting on a chair [[000,111,222,333]]"_   
given a prompt and an image, how should I format these labels? Should I pass them as plain text? Don't I need to transform the bounding boxes into patches first? 

Thank you in advance!
